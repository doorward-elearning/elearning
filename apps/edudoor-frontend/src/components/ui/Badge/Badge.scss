.ed-badge {
  background: var(--accent);
  color: var(--text-alternate);
  padding: 0 var(--padding);
  border-radius: 10px;
}
