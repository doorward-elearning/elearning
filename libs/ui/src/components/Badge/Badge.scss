.ed-badge {
  background: var(--accent);
  color: var(--text-alternate);
  padding: var(--padding-sm) var(--padding);
  border-radius: 5px;
}
