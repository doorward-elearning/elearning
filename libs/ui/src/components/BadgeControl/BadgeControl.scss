.ed-badge-control {
  position: relative;
  &__parent {
  }

  &__badge {
    position: absolute;
    top: -10px;
    right: -15px;
    font-size: 0.8em !important;
    padding: 2px !important;
    width: 20px;
    height: 20px;
    display: grid;
    place-items: center;
    border-radius: 50% !important;
    box-shadow: var(--box-shadow);
    color: white !important;
  }
}
