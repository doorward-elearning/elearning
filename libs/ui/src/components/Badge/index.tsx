import React from 'react';
import './Badge.scss';
import classNames from 'classnames';

const Badge: React.FunctionComponent<BadgeProps> = props => {
  return (
    <span
      className={classNames({
        'ed-badge': true,
      })}
    >
      {props.children}
    </span>
  );
};

export interface BadgeProps {}

export default Badge;
