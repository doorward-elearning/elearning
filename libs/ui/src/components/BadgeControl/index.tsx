import React from 'react';
import './BadgeControl.scss';
import Badge from '@doorward/ui/components/Badge';
import Icon from '../Icon';

const BadgeControl: React.FunctionComponent<BadgeControlProps> = (props): JSX.Element => {
  return (
    <div className="ed-badge-control">
      <div className="ed-badge-control_parent">{props.children}</div>
      {props.value ? <Badge className="ed-badge-control__badge">{props.value}</Badge> : null}
      {props.bookmark ? <Icon className="ed-badge-control__bookmarkIcon" icon="bookmark" /> : null}
    </div>
  );
};

export interface BadgeControlProps {
  value: string | number;
  bookmark?: boolean;
}

export default BadgeControl;
