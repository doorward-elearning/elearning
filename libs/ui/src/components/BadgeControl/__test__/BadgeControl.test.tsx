import * as React from 'react';
import { shallow } from 'enzyme';
import BadgeControl from '../index';
import '@testing-library/jest-dom';
import '../Badge.scss';
import classNames from 'classnames';



describe('BadgeControl component test', () => {
  const iconValue = <img alt="Test icon" />;
  // Snapshot Tests
  it('should render a badge control component with value', () => {
    const wrapper = shallow(<BadgeControl value={1} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a badge control component with icon ', () => {
    const wrapper = shallow(<BadgeControl value={''} bookmark={true} />);
    expect(wrapper).toMatchSnapshot();
  });

    //prop Tests
  it('should have the class ed-badge-control ', () => {
    const wrapper = shallow(<BadgeControl value={''} />);
    expect(wrapper.prop('className')).toContain('ed-badge-control');
  });
});


