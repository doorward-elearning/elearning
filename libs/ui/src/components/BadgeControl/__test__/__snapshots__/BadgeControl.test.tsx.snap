// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeControl component test should render a badge control component with icon  1`] = `ShallowWrapper {}`;

exports[`BadgeControl component test should render a badge control component with value 1`] = `ShallowWrapper {}`;
